tests/functional/test_s3.py

```diff
@@ -429,14 +432,18 @@ def stub_get_object(self, full_contents, start_byte=0, end_byte=None):
             }
         )
         expected_params.update({"Bucket": self.bucket, "Key": self.key})
+        if extra_params is not None:
```

Comment:
Kind of a nit here, but can we get rid of this if statement by just defaulting `extra_params` to `{}` instead of `None`?

Reply:
It's not recommended to supply a default argument that's mutable since default arguments are evaluated once. I guess it's not too important since this test should only be run once, but I'd rather not do it at all.

Reply:
The convention for defaulting to `{}` would be, which is just the inverse of what Steve is doing:

```
if extra_params is None:
    extra_params = {}
```

You can't use singletons like lists or dicts in function signatures because they're initialized in global scope and will maintain state between invocations. [This article](https://realpython.com/python-optional-arguments/#data-types-that-shouldnt-be-used-as-default-arguments) gives a decent overview.